packages/web/src/components/premium-content-purchase-modal/components/PurchaseContentFormFields.tsx

```diff
@@ -19,64 +35,119 @@ import { PurchaseSummaryTable } from './PurchaseSummaryTable'
 
 const messages = {
   purchaseSuccessful: 'Your Purchase Was Successful!',
-  manualTransfer: '(Advanced) Manual Crypto Transfer'
+  existingBalance: 'Existing balance',
+  card: 'Add funds with card',
+  manualTransfer: 'Add with crypto transfer',
+  paymentMethod: 'Payment Method',
+  dollarSign: '$'
 }
 
 type PurchaseContentFormFieldsProps = Pick<
   PurchaseContentFormState,
   'purchaseSummaryValues' | 'stage' | 'isUnlocking'
->
+> & { price: number }
 
 export const PurchaseContentFormFields = ({
+  price,
   purchaseSummaryValues,
   stage,
   isUnlocking
 }: PurchaseContentFormFieldsProps) => {
-  const { onOpen: openUsdcManualTransferModal } = useUSDCManualTransferModal()
+  const { data: balance } = useUSDCBalance()
   const payExtraAmountPresetValues = usePayExtraPresets(useRemoteVar)
+  const [{ value: purchaseMethod }, , { setValue: setPurchaseMethod }] =
+    useField(PURCHASE_METHOD)
   const isPurchased = stage === PurchaseContentStage.FINISH
+  const balanceUSDC = USDC((balance ?? new BN(0)) as BN).value
+  const { extraAmount } = usePurchaseSummaryValues({
+    price,
+    currentBalance: balance
+  })
+  const isExistingBalanceDisabled =
```

Comment:
If we're going to need this logic on mobile as well, I might recommend moving it into the utils in common, possibly even into a boolean exposed by `usePurchaseSummaryValues`? The original intention of that hook was to calculate how much the user owes and how much of the existing balance should be applied. So it feels a little strange here to call it and then use some of the information to do a similar check (can we use the balance?) again. I'm also wondering what happens to `usePurchaseSummaryValues` when you change the purchase method to explicitly force using only a card? I think that hook is going to need to go away in favor of something that is more in line with the new flow.

Reply:
yeah i agree.. hook might need to go away.